front-end/src/components/CustomerAdminComponents/CustomerForm.js

```diff
@@ -0,0 +1,58 @@
+import React from "react";
+import ReactForm from "../ReactForm";
+import {toast} from "react-toastify";
+import PersonFields from "../PersonFields";
+
+export default class CustomerForm extends ReactForm {
+  constructor(props) {
+    super(props);
+    if (props.initialCustomer) {
+      const {givenName, familyName, ssn, sin, address} = props.initialCustomer;
+      this.state = {
+        givenName, familyName, ...address,
+        ssn: ssn || "", sin: sin || ""
+      };
+    } else {
+      this.state = {
+        givenName: "",
+        familyName: "",
+        ssn: "",
+        sin: "",
+        streetName: "",
+        streetNumber: "",
+        city: "",
+        country: ""
+      }
+    }
+  }
+
+  onSubmit = async event => {
```

Comment:
This try catch duplicates the catch in the parent (modal). The current pattern in use is that the forms don't catch errors. Additionally the sin / ssn fields can be simplified to:
`sin: sin || null`